[fix #7] [mozmeao/infra#1104] Clean up oregon-a dev yaml.

jpetto opened the pull request.

jpetto committed:
[fix #7] [mozmeao/infra#1104] Clean up oregon-a dev yaml.

jgmize commented on oregon-a/basket-dev/ns.yaml:
```diff
@@ -1,13 +1,7 @@
-  uid: 645beaf8-2223-11e8-8262-02e6188a289c
 spec:
   finalizers:
   - kubernetes
```
I don't believe we need the spec or status sections either, based on https://kubernetes.io/docs/tasks/administer-cluster/namespaces/#creating-a-new-namespace

jgmize commented on oregon-a/basket-dev/web-deploy.yaml:
```diff
@@ -48,366 +37,226 @@ spec:
-            secretKeyRef:
-              key: deis-app
-              name: basket-dev-v107-env
-        - name: DEIS_DOMAIN
```
This and the DEIS_APP deletion above are blocked by https://github.com/mozmeao/basket/issues/206

jpetto replied:
In which order do you think we should make these changes? Or should we try to coordinate them as close together as possible?

jgmize replied:
On further review, I don't believe the middleware update is a blocker for removing the DEIS_* values. As far as I can tell the missing env vars shouldn't break anything in the [current code](https://github.com/mozmeao/basket/blob/2a653b4e3a67e55b7628289dc86e2bc7cdacbda5/basket/news/middleware.py#L33-L43), the only impact should be that `X-Backend-Server` will only have the HOSTNAME value.

jgmize commented on oregon-a/basket-dev/web-deploy.yaml:
```diff
@@ -48,366 +37,226 @@ spec:
-            secretKeyRef:
-              key: email-port
-              name: basket-dev-v107-env
+          value: 587
```
I believe leaving the quotes off of this number will result in a type error, since an env var need to be a string

jgmize commented on oregon-a/basket-dev/web-deploy.yaml:
```diff
@@ -48,366 +37,226 @@ spec:
+          value: "True"
+        - name: WSGI_NUM_WORKERS
+          value: 4
+        image: mozmeao/basket:4d0668d
```
All of the current [tags](https://hub.docker.com/r/mozorg/basket/tags) are the full length instead of the short tag.

jgmize replied:
```suggestion
        image: mozmeao/basket:4d0668d4e5cb304e68b04bcc0d2b278d27b591e6
```

jgmize commented on oregon-a/basket-dev/worker-deploy.yaml:
```diff
@@ -48,366 +37,226 @@ spec:
-            secretKeyRef:
-              key: new-relic-app-name
-              name: basket-dev-v107-env
+          value: basket-dev-oregon-b
```
```suggestion
          value: basket-dev-oregon-a;basket-dev
```

jpetto replied:
Assuming this change should also be made to `web-deploy.yaml`?

jgmize commented on oregon-a/basket-dev/worker-deploy.yaml:
```diff
@@ -48,366 +37,226 @@ spec:
+          value: "True"
+        - name: WSGI_NUM_WORKERS
+          value: 4
+        image: mozmeao/basket:4d0668d
```
```suggestion
        image: mozmeao/basket:4d0668d4e5cb304e68b04bcc0d2b278d27b591e6
```

jgmize commented on oregon-a/basket-dev/worker-deploy.yaml:
```diff
@@ -48,366 +37,226 @@ spec:
-        image: 127.0.0.1:5555/basket-dev:v107
+          value: "True"
+        - name: WSGI_NUM_WORKERS
+          value: 4
```
```suggestion
          value: "4"
```

To prevent a type error

jpetto committed:
Review fixes:

- remove superfluous info from ns.yaml
- quote all numeric values in env vars
- use full image hash
